internal/configs/configload/loader_load.go

```diff
@@ -24,14 +24,14 @@ import (
 //
 // LoadConfig performs the basic syntax and uniqueness validations that are
 // required to process the individual modules
-func (l *Loader) LoadConfig(rootDir string) (*configs.Config, hcl.Diagnostics) {
-	return l.loadConfig(l.parser.LoadConfigDir(rootDir))
+func (l *Loader) LoadConfig(rootDir string, call configs.StaticModuleCall) (*configs.Config, hcl.Diagnostics) {
```

Comment:
Can I ask what the resolution was here? I would've preferred a pointer here as it will help set expectations for devs working in this area of code about how things are passed around (and if they have to check for empty/nil)

Reply:
configs.StaticModuleCall is always required and not optional